src/lib/hash.ts

```diff
@@ -192,3 +194,40 @@ class TokenSymbol extends Struct(TokenSymbolPure) {
 function emptyReceiptChainHash() {
   return emptyHashWithPrefix('CodaReceiptEmpty');
 }
+
+function buildSHA(length: 224 | 256 | 384 | 512, nist: boolean) {
+  return {
+    hash(message: (Field | UInt8)[]) {
+      if (message.length === 0) {
+        throw Error('SHA hash of empty message');
+      }
+
+      const values = message.map((f) => {
+        if (isField(f)) {
+          // Make sure that the field is exactly a byte.
+          f.toBits(8);
+          return f.value;
+        }
+        return f.value.value;
+      });
+
+      return Snarky.sha.create([0, ...values], nist, length).map(Field);
+    },
+  };
+}
+
+const Hash = {
+  default: Poseidon.hash,
+
+  Poseidon: Poseidon.hash,
+
+  SHA224: buildSHA(224, true).hash,
```

Comment:
nit: we can make `buildSHA` return the function directly, rather than an object containing a single `hash` function - I'll change that in the RFC if we decide to go this route